src/rt/util/hash.d

```diff
@@ -17,22 +17,22 @@ version( AnyX86 )
     version = HasUnalignedOps;
 
 
-@trusted pure nothrow
+@trusted pure nothrow @nogc
 size_t hashOf( const(void)[] buf, size_t seed = 0 )
 {
     return hashOf(buf.ptr, buf.length, seed);
 }
 
-@trusted pure nothrow
-size_t hashOf( const(void)* buf, size_t len, size_t seed = 0 )
+@trusted pure nothrow @nogc
```

Comment:
I left the old function in there because removing it may break existing code, adding system to it may break existing code, and breaking existing code should be done separately if possible. Not lumped in with other things. Even if `hashOf` is an internal function, there's nothing stopping anyone from using it. There being another `hashOf` in object.d is unfortunate and further obfuscates things, but again, that is not the topic of this PR.